dask/dataframe/core.py

```diff
@@ -4939,3 +4939,137 @@ def meta_warning(df):
                 "  Before: .apply(func)\n"
                 "  After:  .apply(func, meta=%s)\n" % str(meta_str))
     return msg
+
+
+def prefix_reduction(f, ddf, identity=None, preprocess=None, postprocess=None):
+    """ Computes the prefix sums of f on df
+
+    If df has partitions [P1, P2, ..., Pn], then returns the DataFrame with
+    partitions [f(z, P1), f(f(z, P1), P2), f(f(f(z, P1), P2), P3), ...]
+
+    If you don't specify identity, make sure f handles when either argument is
+    None!
+
+    Parameters
+    ----------
+    f : 'a x 'a -> 'a
```

Comment:
Is there any appropriate place for me to specify the typing so that it's clear to users? Or should I just leave it out and have them figure it out themselves?

Reply:
In the description is fine.

Reply:
To be clear, you should do something like this:

```
f : callable
    An associative function of the form ...
```